server/src/main/java/org/apache/druid/client/indexing/HttpIndexingServiceClient.java

```diff
@@ -67,7 +67,8 @@ public HttpIndexingServiceClient(
   @Override
   public void killUnusedSegments(String dataSource, Interval interval)
   {
-    runTask(new ClientKillUnusedSegmentsTaskQuery(dataSource, interval));
+    final ClientTaskQuery taskQuery = new ClientKillUnusedSegmentsTaskQuery(null, dataSource, interval);
+    runTask(taskQuery.getId(), taskQuery);
```

Comment:
Do we pull the ID from the `taskQuery` anywhere except here? If not, maybe we could just pass in the ID here without adding an ID field to the task query.

Reply:
`ClientTaskQuery` is used by the coordinator when it issues kill tasks or compaction tasks. The coordinator cannot use `Task` directly because of a dependency issue, and so `ClientTaskQuery` is supposed to be equivalent to `Task`. The `id` of `taskQuery` should be same with the `id` of `task` for that reason after it is deserialized in the overlord. I think we still need an ID field for `ClientTaskQuery`.

However, I like the idea of passing in the ID. When the coordinator issues a task, we can generate an ID distinguishable from those issued by users. After this PR, the ID of compaction and kill tasks will be prefixed by `coordinator-issued` if they are submitted by coordinator. `api-issued` prefix will be used when a kill task is issued by calling `DataSourcesResource.killUnusedSegmentsInInterval()`.